components/usage/pkg/stripe/stripe.go

```diff
@@ -122,6 +137,47 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 	}, nil
 }
 
+// GetUpcomingInvoice fetches the upcoming invoice for the given team or user id.
+func (c *Client) GetUpcomingInvoice(ctx context.Context, in *v1.GetUpcomingInvoiceRequest) (*StripeInvoice, error) {
+	var query = ""
+	if in.GetTeamId() != "" {
+		query = fmt.Sprintf("metadata['teamId']:'%s'", in.GetTeamId())
+	}
+	if in.GetTeamId() != "" {
+		query = fmt.Sprintf("metadata['userId']:'%s'", in.GetUserId())
+	}
+	searchParams := &stripe.CustomerSearchParams{
+		SearchParams: stripe.SearchParams{
+			Query:   query,
+			Expand:  []*string{stripe.String("data.subscriptions")},
+			Context: ctx,
+		},
+	}
+	iter := c.sc.Customers.Search(searchParams)
+	if iter.Err() != nil && !iter.Next() {
+		return nil, ErrorCustomerNotFound
+	}
+	customer := iter.Customer()
```

Comment:
What if the iterator contains multiple results?

Reply:
raising then